#1. 이미지 읽기

In [13]:
from google.colab import drive

drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


##1.1 이미지 읽기

In [ ]:
import cv2

img_path = '/content/drive/MyDrive/16. 이미지 분석/img/img01.jpg'

image = cv2.imread(img_path)

image

##1.2 이미지 정보 확인

In [ ]:
image.shape

(1440, 1920, 3)

>image.shape의 실행 결과 1440은 height, 1920은 width, 3은 channels을 의미하며 channels는 RGB 색상을 뜻한다.

In [ ]:
height, width, channels = image.shape

print('이미지 가로 : ', width)
print('이미지 세로 : ', height)
print('이미지 채널 : ', channels)

이미지 가로 :  1920
이미지 세로 :  1440
이미지 채널 :  3


##1.3 이미지 출력

In [ ]:
import cv2

img_path = '/content/drive/MyDrive/16. 이미지 분석/img/img01.jpg'

image = cv2.imread(img_path)

cv2.imshow(image)

DisabledFunctionError: ignored

In [ ]:
import cv2
from matplotlib import pyplot as plt

#Import image
image = cv2.imread(img_path)

#Show the image with matplotlib
plt.imshow(image)
plt.show()

In [ ]:
import cv2
from google.colab.patches import cv2_imshow

cvim2disp = cv2.imread(img_path)
cv2_imshow(cvim2disp)

#2. 동영상 불러오기

##2.1 동영상 정보 확인

In [ ]:
import cv2

movie_path='/content/drive/MyDrive/16. 이미지 분석/mov/mov01.avi'
mov = cv2.VideoCapture(movie_path)

width = mov.get(cv2.CAP_PROP_FRAME_WIDTH)
height = mov.get(cv2.CAP_PROP_FRAME_HEIGHT)
count = mov.get(cv2.CAP_PROP_FRAME_COUNT)
fps = mov.get(cv2.CAP_PROP_FPS)

print(width, ' : ', height)
print('총 frame 수 : ', count)
print('초당 frame 수 : ', fps)

1920.0  :  1440.0
총 frame 수 :  401.0
초당 frame 수 :  30.0


##2.2 동영상 frame 출력

In [ ]:
import tensorflow as tf

device_name = tf.test.gpu_device_name()

if device_name != '/device:GPU:0':
  raise SystemError('GPU device not found')

print('Found GPU at : ', device_name)

Found GPU at :  /device:GPU:0


In [14]:
import cv2

movie_path='/content/drive/MyDrive/16. 이미지 분석/mov/mov01.avi'
mov = cv2.VideoCapture(movie_path)
mov

<VideoCapture 0x7f53486f39d0>

In [ ]:
from google.colab.patches import cv2_imshow

# with tf.device('/device:GPU:0'):
with tf.device(device_name):
  # while mov.isOpened():
  for i in range(10):
    mov.isOpened()

    isRead, frame = mov.read()

    if isRead:
      cv2_imshow(frame)

  mov.release()
  cv2.destroyAllWindows()

In [ ]:
from google.colab.patches import cv2_imshow


for i in range(10):
  # mov.isOpened()

  isRead, frame = mov.read()

  if isRead:
    cv2_imshow(frame)

mov.release()
cv2.destroyAllWindows()

#3. 동영상을 이미지로 저장하기

In [17]:
import cv2

path = '/content/drive/MyDrive/16. 이미지 분석/'
movie_path = path + 'mov/mov01.avi'
mov = cv2.VideoCapture(movie_path)

for i in range(10):
  mov.isOpened()
  isRead, frame = mov.read()

  if isRead:
    # cv2_imshow(frame)
    file_path = path + 'snapshot/snapshot_'+str(i)+'.jpg'
    cv2.imwrite(file_path, frame)

mov.release()
cv2.destroyAllWindows()

#4. 저장된 frame 출력

In [ ]:
import cv2

path = '/content/drive/MyDrive/16. 이미지 분석/'

for i in range(10):
  file_path = path + 'snapshot/snapshot_'+str(i)+'.jpg'
  image = cv2.imread(file_path)
  cv2_imshow(image)